### ISSUE TITLE
What is the right hyper parameter for mobilenet V2 to get 94.47% acc on cifar10?

### ISSUE THREAD

**Issue by birdyLinch** [kuangliu/pytorch-cifar#29]
The first time I run mobilenet v2 experiment, it gives me an error in forward function. So I change the the kernel size of avg pooling layer from 2 --> 4. I train this model using batch size 128, rl 0.1 (x0.1 when epoch in [150, 250]). It gives me around 91.0% .  What could be the problem? Thanks!

**Comment by kuangliu** [comment]
It is 4, my mistake. And it's 94.43% not 94.47%.
I trained MobileNetV2 for nearly 300 epochs with lr 0.001 (starting from 0.1). 
Just try again.

**Comment by czhu95** [comment]
Hi kuangliu,

I am also trying to reproduce your mobilenet v2 results. May I ask your learning rate decay strategy? I trained mobilenet v2 starting at lr = 0.1 and decayed at epoch 150 and 250. I trained 600 epochs in total but still can only reach 91% top 1 acc.

Could you please help me with this? Thank you very much!

**Comment by kilho** [comment]
I also get 91% top 1 accuracy with lr schedule mentioned in this git.

When I run 500 epochs with initial lr =0.1 and [100, 200, 300, 400] update with 0.1 gamma, I got around ~ 92%

https://github.com/MG2033/MobileNet-V2
has implemented MobileNetV2 and got 90.9 %. (I did not checked if they are the same version).

**Comment by zhaohui-yang** [comment]
Just set weight decay to be 4e-5, same with paper, this would improve top1 to around 94%

**Comment by thanhmvu** [comment]
Thanks @zhaohui-yang for the tip. Just want to confirm that with everything else the same (sgd, epochs 350, batch size 128, lr 0.1, steps [150, 250]), I got (best results):
- Epoch 282 | Loss: 0.263 | Acc: **91.740%** (9174/10000) for weight decay **5e-4**
- Epoch 348 | Loss: 0.316 | Acc: **94.200%** (9420/10000) for weight decay **4e-5**
